packages/core/src/messaging/eventBus.ts

```diff
@@ -0,0 +1,46 @@
+import { EventEmitter } from "eventemitter3";
+import type { Logger } from "../logging/logger.ts";
+
+export type EventName = string | symbol;
+
+export type EventCallbackFunction = (data?: unknown) => void;
+
+export interface EventBusOptions {
+    logger?: Logger;
+}
+
+export interface AddListenerOptions {
+    once?: boolean;
+}
+
+export interface RemoveListenerOptions {
+    once?: boolean;
+}
+
+export class EventBus {
+    private _eventEmitter: EventEmitter;
+    private _logger?: Logger;
+
+    constructor({ logger }: EventBusOptions = {}) {
+        this._eventEmitter = new EventEmitter();
+        this._logger = logger;
+    }
+
+    addListener(eventName: EventName, callback: EventCallbackFunction, { once }: AddListenerOptions = {}) {
+        if (once === true) {
+            this._eventEmitter.once(eventName, callback);
+        } else {
+            this._eventEmitter.addListener(eventName, callback);
+        }
+    }
+
+    removeListener(eventName: EventName, callback: EventCallbackFunction, { once }: RemoveListenerOptions = {}) {
+        this._eventEmitter.removeListener(eventName, callback, once);
+    }
+
+    dispatch(eventName: EventName, data?: unknown) {
```

Comment:
i can look into this after, but eventEmitter seems to have all the data, the EventCallbackFunction typed, and we put unknown everywhere. I feel like we could reuse their typings

Reply:
Yeah if you can improve this that would be great!

Maybe you could do it in a distinct PR?

Reply:
Resolving as I added an issue: https://github.com/workleap/wl-squide/issues/7